src/backend/opencl/sparse_arith.cpp

```diff
@@ -120,11 +163,14 @@ SparseArray<T> arithOpS(const SparseArray<T> &lhs, const Array<T> &rhs, const bo
                                                   const bool reverse);                              \
     template SparseArray<T> arithOpS<T, af_div_t>(const SparseArray<T> &lhs, const Array<T> &rhs,   \
```

Comment:
Can't these be specialization of the arithOp fuction? I don't see why you need a different name suffix at the end.

Reply:
I am not sure how that is possible, the second parameter to `arithOpS` is a normal `Array`, not `SparseArray`.

Reply:
its basically overloading right?